### PR TITLE
remove Child.ref

### PR DESCRIPTION
**Pull request by mat-hek**

### CODE REVIEW

File: lib/membrane/core/parent/child_life_controller/startup_utils.ex [hunk 1]
```diff
@@ -123,15 +123,15 @@ defmodule Membrane.Core.Parent.ChildLifeController.StartupUtils do
...
     new_children_groups =
       Enum.map(children_definitions, fn {_children, options} -> options.group end)
 
     state_children_names =
       Map.keys(state.children)
-      |> Enum.map(fn child_ref ->
-        case child_ref do
+      |> Enum.map(fn child_name ->
+        case child_name do
           {Membrane.Child, _group, child_name} -> child_name
```
Comment (FelonEkonom) [inline lib/membrane/core/parent/child_life_controller/startup_utils.ex:135]: ` {Membrane.Child, _group, child_name}` - this is equal `Child.ref(child_name, group: _groups`, so it should be removed

File: lib/membrane/core/parent/child_life_controller/startup_utils.ex [hunk 2]
```diff
@@ -175,8 +175,8 @@ defmodule Membrane.Core.Parent.ChildLifeController.StartupUtils do
   end
 
   defp get_children_names(children) do
-    Enum.map(children, fn {child_ref, _child_module, _options} ->
-      case child_ref do
+    Enum.map(children, fn {child_name, _child_module, _options} ->
+      case child_name do
         {Membrane.Child, _group, child_name} -> child_name
```
Comment (FelonEkonom) [inline lib/membrane/core/parent/child_life_controller/startup_utils.ex:180]: as above

File: lib/membrane/element/base.ex [hunk 3]
```diff
@@ -237,7 +237,6 @@ defmodule Membrane.Element.Base do
     bring_child =
       if Keyword.get(options, :bring_child?, true) do
         quote do
-          require Membrane.Child, as: Child
         end
       end
```
Comment (FelonEkonom) [inline lib/membrane/element/base.ex:241]: remove whole `bring_child` variable

File: lib/membrane/pipeline.ex [hunk 4]
```diff
@@ -459,7 +459,6 @@ defmodule Membrane.Pipeline do
     bring_child =
       if Keyword.get(options, :bring_child?, true) do
         quote do
-          require Membrane.Child, as: Child
         end
       end
```
Comment (FelonEkonom) [inline lib/membrane/pipeline.ex:463]: as above

File: test/membrane/integration/child_spawn_test.exs [hunk 5]
```diff
@@ -200,25 +200,25 @@ defmodule Membrane.Integration.ChildSpawnTest do
...
+  #     |> child(:sink, Testing.Sink)
+
+  #   Testing.Pipeline.execute_actions(pipeline_pid, spec: spec)
+  #   assert_receive {:DOWN, ^pipeline_ref, :process, ^pipeline_pid, {reason, _stack_trace}}
+
+  #   assert reason.message =~
+  #            ~r/Improper name: {Membrane.Child, :first_group, :source}/
+
+  #   Testing.Pipeline.terminate(pipeline_pid)
+  # end
```
Comment (FelonEkonom) [inline test/membrane/integration/child_spawn_test.exs:223]: Remove commented lines

**Comment by FelonEkonom** [comment]
Ensure, that you check if a child name is unique and update CHANGELOG.md and upgrading guide

File: test/membrane/integration/child_spawn_test.exs [hunk 6]
```diff
@@ -200,25 +200,25 @@ defmodule Membrane.Integration.ChildSpawnTest do
...
-
-    Testing.Pipeline.execute_actions(pipeline_pid, spec: spec)
-    assert_receive {:DOWN, ^pipeline_ref, :process, ^pipeline_pid, {reason, _stack_trace}}
-
-    assert reason.message =~
-             ~r/Improper name: {Membrane.Child, :first_group, :source}/
-
-    Testing.Pipeline.terminate(pipeline_pid)
-  end
+  # test "if the pipeline raises an exception when there is an attempt to spawn a child with a name satisfying the Membrane's reserved pattern" do
```
Comment (varsill) [inline test/membrane/integration/child_spawn_test.exs:203]: another commented out test, I don't think there are other reserved Membrane's patterns that could be caught in this place, so we can probably remove it

**Review by varsill — CHANGES_REQUESTED** [review]

**Review by varsill — COMMENTED** [review]
There is still a name "children_refs" here: https://github.com/membraneframework/membrane_core/blob/7e9f77fc1fcf20dcaf4a0f9e332786c562ff8c46/lib/membrane/core/parent/child_life_controller.ex#L508

**Review by varsill — APPROVED** [review]

**Pull request merged**